### DISCUSSION TITLE
How to reliably get the target triple for autoconf

### DISCUSSION THREAD

**Discussion by bmerry** [cross-rs/cross#1174] [Q&A] [answered]
I'm using a pre-build script to install a C library (libpcap) to link against. The library uses autoconf for configuration, so to cross-compile it I need to run `./configure --host=<triple>`, where `<triple>` needs to match the prefix for the C compiler e.g. `arm-linux-musleabihf`. Ideally I'd like to compute this rather than needing to hard-code it for each architecture.

I'm currently using `$(dpkg-architecture -A $CROSS_DEB_ARCH -q DEB_TARGET_GNU_TYPE)` to get the target triple, but it feels inelegant given that cross is already populating environment variables with the compiler path. I've also thought about rummaging around the environment variables to figure out what cross has set up and then strip the `-gcc` off the end, but that seems even hackier.

Is there a clean way to obtain the target triple to pass to autoconf?

**Comment by Emilgardis** [comment, marked as answer]
if you're using the `main` images, you can use [$CROSS_TOOLCHAIN_PREFIX](https://github.com/cross-rs/cross/blob/f0190b0c0fd8fc354d9b0fad61c06bf0f29be277/docker/Dockerfile.aarch64-unknown-linux-gnu#LL34C5-L34C27), and strip of the trailing `-`

**Reply by bmerry** [reply]
Thanks, I guess I'll wait until that ends up in a release. It definitely seems much easier, particularly if all the images use bash (`${CROSS_TOOLCHAIN_PREFIX%-}`).
